lib/fetcher.js

```diff
@@ -101,13 +102,32 @@ module.exports = heroku => {
     return attachments.filter(a => a.addon.plan.name.startsWith('heroku-postgresql'))
   }
 
+  function * getAttachmentNamesByAddons (attachments) {
+    let results = {}
+    attachments.forEach(function (a) {
+      let attachmentNames = results[a.addon.id] || []
+      attachmentNames = attachmentNames.concat(a.name)
+      results[a.addon.id] = attachmentNames
+    })
+
+    return results
+  }
+
   function * all (app) {
     const uniqby = require('lodash.uniqby')
 
     debug(`fetching all DBs on ${app}`)
 
-    let addons = (yield allAttachments(app)).map(a => a.addon)
+    let attachments = yield allAttachments(app)
+    let addons = attachments.map(a => a.addon)
+
+    // Get the list of config vars per addon here and add to each addon
+    let attachmentNamesByAddons = yield getAttachmentNamesByAddons(attachments)
     addons = uniqby(addons, 'id')
+    addons = addons.map(function (a) {
```

Comment:
I'd chain this call with the `uniqby` above--`addons = unqiby(addons, 'id').map...`.